src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -193,6 +198,64 @@ public void deleteRoles(final Node node) throws RepositoryException {
     };
 
     /**
+     *  This maps a Collection of acl:agentClass values to a List of agents.
+     *  Any out-of-domain URIs are silently ignored.
+     */
+    private List<String> dereferenceAgentClasses(final Collection<String> agentClasses) {
+        final Session internalSession = sessionFactory.getInternalSession();
+        final IdentifierConverter<Resource, FedoraResource> translator =
+                new DefaultIdentifierTranslator(internalSession);
+
+        final List<String> members = new ArrayList<>();
+        agentClasses.stream()
+                    .distinct()
+                    .forEach(x -> {
+                        if (x.startsWith(FEDORA_INTERNAL_PREFIX)) {
+                            final FedoraResource resource = nodeService.find(
+                                internalSession, x.substring(FEDORA_INTERNAL_PREFIX.length()));
+                            members.addAll(getFoafMembers(translator, resource));
+                        } else if (x.startsWith(FOAF_NAMESPACE_VALUE)) {
```

Comment:
Where does this logic come from? Does it map into the "spec"?
https://www.w3.org/wiki/WebAccessControl